core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -215,21 +227,20 @@ void setKeyRefer(int entryIndex, OakRKeyReference keyReferBuffer) {
         if (entryIndex == Chunk.NONE) {
             return;
         }
-        int blockID = getEntryField(entryIndex, OFFSET_KEY_BLOCK);
-        int keyPosition = getEntryField(entryIndex, OFFSET_KEY_POSITION);
-        int length = getEntryField(entryIndex, OFFSET_KEY_LENGTH);
+        int blockID = getEntryField(entryIndex, OFFSET.KEY_BLOCK);
+        int keyPosition = getEntryField(entryIndex, OFFSET.KEY_POSITION);
+        int length = getEntryField(entryIndex, OFFSET.KEY_LENGTH);
         keyReferBuffer.setKeyReference(blockID, keyPosition, length);
     }
 
     /**
      * release key in slice, currently not in use, waiting for GC to be arranged
      **/
     void releaseKey(int entryIndex) {
-        int blockID = getEntryField(entryIndex, OFFSET_KEY_BLOCK);
-        int keyPosition = getEntryField(entryIndex, OFFSET_KEY_POSITION);
-        int length = getEntryField(entryIndex, OFFSET_KEY_LENGTH);
-        ByteBuffer bb = memoryManager.getByteBufferFromBlockID(blockID, keyPosition,length);
-        Slice s = new Slice(blockID, bb);
+        int blockID = getEntryField(entryIndex, OFFSET.KEY_BLOCK);
```

Comment:
If we have here all those writeKey/readKey/setKeyRef/releaseKey AND values are now treated similarly as a keys, isn't it reasonable to have near writeValue/readValue/setValueRef/releaseValue or even same function to deal with key/value?

Reply:
There is still a difference between keys and values.
First, keys do not change and thus reading them can be does non-atomically (should be kept that way to avoid unnecessary overhead when performing a lookup).
Second, values have a header to initialize and keys do not, and value are not directly written to an entry and keys do.

Regarding setKeyRef, I'm not sure how to use it for values.

I added releaseValue for consistency sake.

Reply:
OK